app/models/invite_code_validator.rb

```diff
@@ -0,0 +1,29 @@
+class InviteCodeValidator < ActiveModel::Validator
+  def validate(user)
+
+    user_invite_code = InviteCode.find_by_invite_code user.invite_code
```

Comment:
I had another look at the `find_by_invite_code(nil)` returning the first record in the table. Turns out this was a couchrest model bug. Luckily it's fixed in the new version. #195 will fix this for leap_web.
So we can safely rely on this to not return an invite code even if the user somehow manages to make user.invite_code return nil. :sweat_smile: